Fix `DropwizardSSLConnectionSocketFactoryTest` on Java 17

`DropwizardSSLConnectionSocketFactoryTest` had to be disabled on Java 17.

```text
ERROR [2021-09-24 09:13:39,834] io.dropwizard.cli.ServerCommand: Unable to start server, shutting down
! java.io.IOException: exception decrypting data - javax.crypto.BadPaddingException: pad block corrupted
! at org.bouncycastle.jcajce.provider.keystore.pkcs12.PKCS12KeyStoreSpi.cryptData(Unknown Source)
! at org.bouncycastle.jcajce.provider.keystore.pkcs12.PKCS12KeyStoreSpi.engineLoad(Unknown Source)
! at org.bouncycastle.jcajce.provider.keystore.util.AdaptingKeyStoreSpi.engineLoad(Unknown Source)
! at java.base/java.security.KeyStore.load(KeyStore.java:1473)
! at org.eclipse.jetty.util.security.CertificateUtils.getKeyStore(CertificateUtils.java:54)
```

<details>
<summary>Full error message in dropwizard-client</summary

```text
[INFO] Running io.dropwizard.client.DropwizardSSLConnectionSocketFactoryTest
INFO  [2021-09-24 09:13:39,151] org.eclipse.jetty.util.log: Logging initialized @2878ms to org.eclipse.jetty.util.log.Slf4jLog
INFO  [2021-09-24 09:13:39,250] io.dropwizard.server.DefaultServerFactory: Registering jersey handler with root path prefix: /
INFO  [2021-09-24 09:13:39,252] io.dropwizard.server.DefaultServerFactory: Registering admin handler with root path prefix: /
INFO  [2021-09-24 09:13:39,257] io.dropwizard.server.ServerFactory: Starting TlsTestApplication
INFO  [2021-09-24 09:13:39,402] org.eclipse.jetty.setuid.SetUIDListener: Opened application@4a225014{SSL, (ssl, http/1.1)}{0.0.0.0:62150}
INFO  [2021-09-24 09:13:39,403] org.eclipse.jetty.setuid.SetUIDListener: Opened application@7d836c4a{SSL, (ssl, http/1.1)}{0.0.0.0:62151}
INFO  [2021-09-24 09:13:39,403] org.eclipse.jetty.setuid.SetUIDListener: Opened application@485547ac{SSL, (ssl, http/1.1)}{0.0.0.0:62152}
INFO  [2021-09-24 09:13:39,403] org.eclipse.jetty.setuid.SetUIDListener: Opened application@6d77d269{SSL, (ssl, http/1.1)}{0.0.0.0:62153}
INFO  [2021-09-24 09:13:39,404] org.eclipse.jetty.setuid.SetUIDListener: Opened application@350d9d23{SSL, (ssl, http/1.1)}{0.0.0.0:62154}
INFO  [2021-09-24 09:13:39,405] org.eclipse.jetty.setuid.SetUIDListener: Opened application@530a46f4{SSL, (ssl, http/1.1)}{0.0.0.0:62155}
INFO  [2021-09-24 09:13:39,405] org.eclipse.jetty.setuid.SetUIDListener: Opened admin@3180131e{HTTP/1.1, (http/1.1)}{0.0.0.0:62156}
INFO  [2021-09-24 09:13:39,410] org.eclipse.jetty.server.Server: jetty-9.4.43.v20210629; built: 2021-06-30T11:07:22.254Z; git: 526006ecfa3af7f1a27ef3a288e2bef7ea9dd7e8; jvm 17+35-LTS
INFO  [2021-09-24 09:13:39,480] org.eclipse.jetty.util.ssl.SslContextFactory: x509=X509@1b84d03d(1,h=[localhost],a=[],w=[]) for Server@41d8ac75[provider=null,keyStore=file:///Users/joschi/src/dropwizard/dropwizard-client/target/test-classes/stores/server/keycert.p12,trustStore=null]
INFO  [2021-09-24 09:13:39,564] io.dropwizard.jetty.HttpsConnectorFactory: Enabled protocols: [TLSv1.2]
INFO  [2021-09-24 09:13:39,564] io.dropwizard.jetty.HttpsConnectorFactory: Disabled protocols: [SSLv2Hello, SSLv3, TLSv1, TLSv1.1, TLSv1.3]
INFO  [2021-09-24 09:13:39,564] io.dropwizard.jetty.HttpsConnectorFactory: Enabled cipher suites: [TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_DHE_DSS_WITH_AES_128_CBC_SHA256, TLS_DHE_DSS_WITH_AES_128_GCM_SHA256, TLS_DHE_DSS_WITH_AES_256_CBC_SHA256, TLS_DHE_DSS_WITH_AES_256_GCM_SHA384, TLS_DHE_RSA_WITH_AES_128_CBC_SHA256, TLS_DHE_RSA_WITH_AES_128_GCM_SHA256, TLS_DHE_RSA_WITH_AES_256_CBC_SHA256, TLS_DHE_RSA_WITH_AES_256_GCM_SHA384, TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384, TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384, TLS_EMPTY_RENEGOTIATION_INFO_SCSV]
INFO  [2021-09-24 09:13:39,564] io.dropwizard.jetty.HttpsConnectorFactory: Disabled cipher suites: [TLS_DHE_DSS_WITH_AES_128_CBC_SHA, TLS_DHE_DSS_WITH_AES_256_CBC_SHA, TLS_DHE_RSA_WITH_AES_128_CBC_SHA, TLS_DHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDH_RSA_WITH_AES_128_CBC_SHA, TLS_ECDH_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_256_CBC_SHA256, TLS_RSA_WITH_AES_256_GCM_SHA384]
INFO  [2021-09-24 09:13:39,585] org.eclipse.jetty.util.ssl.SslContextFactory: x509=X509@13288174(1,h=[localhost],a=[],w=[]) for Server@24ae2183[provider=null,keyStore=file:///Users/joschi/src/dropwizard/dropwizard-client/target/test-classes/stores/server/self_sign_keycert.p12,trustStore=null]
INFO  [2021-09-24 09:13:39,609] org.eclipse.jetty.util.ssl.SslContextFactory: x509=X509@4accc3de(1,h=[localhost],a=[],w=[]) for Server@1a23136f[provider=null,keyStore=file:///Users/joschi/src/dropwizard/dropwizard-client/target/test-classes/stores/server/keycert.p12,trustStore=file:///Users/joschi/src/dropwizard/dropwizard-client/target/test-classes/stores/server/ca_truststore.ts]
INFO  [2021-09-24 09:13:39,645] org.eclipse.jetty.util.ssl.SslContextFactory: x509=X509@11e75942(1,h=[badhost],a=[],w=[]) for Server@23c61c1b[provider=null,keyStore=file:///Users/joschi/src/dropwizard/dropwizard-client/target/test-classes/stores/server/bad_host_keycert.p12,trustStore=null]
INFO  [2021-09-24 09:13:39,678] org.eclipse.jetty.util.ssl.SslContextFactory: x509=X509@6edb093f(1,h=[localhost],a=[],w=[]) for Server@7f5c4ff8[provider=null,keyStore=file:///Users/joschi/src/dropwizard/dropwizard-client/target/test-classes/stores/server/keycert.p12,trustStore=null]
INFO  [2021-09-24 09:13:39,689] org.eclipse.jetty.util.ssl.SslContextFactory: No Protocols matching 'SSLv1' is supported
INFO  [2021-09-24 09:13:39,689] org.eclipse.jetty.util.ssl.SslContextFactory: No Protocols matching 'SSLv2' is supported
WARN  [2021-09-24 09:13:39,689] org.eclipse.jetty.util.ssl.SslContextFactory: No selected Protocols from [TLSv1.3, TLSv1.2, TLSv1.1, TLSv1, SSLv3, SSLv2Hello]
ERROR [2021-09-24 09:13:39,834] io.dropwizard.cli.ServerCommand: Unable to start server, shutting down
! java.io.IOException: exception decrypting data - javax.crypto.BadPaddingException: pad block corrupted
! at org.bouncycastle.jcajce.provider.keystore.pkcs12.PKCS12KeyStoreSpi.cryptData(Unknown Source)
! at org.bouncycastle.jcajce.provider.keystore.pkcs12.PKCS12KeyStoreSpi.engineLoad(Unknown Source)
! at org.bouncycastle.jcajce.provider.keystore.util.AdaptingKeyStoreSpi.engineLoad(Unknown Source)
! at java.base/java.security.KeyStore.load(KeyStore.java:1473)
! at org.eclipse.jetty.util.security.CertificateUtils.getKeyStore(CertificateUtils.java:54)
! at org.eclipse.jetty.util.ssl.SslContextFactory.loadKeyStore(SslContextFactory.java:1203)
! at org.eclipse.jetty.util.ssl.SslContextFactory.load(SslContextFactory.java:322)
! at org.eclipse.jetty.util.ssl.SslContextFactory.doStart(SslContextFactory.java:244)
! at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
! at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
! at org.eclipse.jetty.server.Server.start(Server.java:423)
! at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
! at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
! at org.eclipse.jetty.server.Server.doStart(Server.java:387)
! at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
! at io.dropwizard.cli.ServerCommand.run(ServerCommand.java:53)
! at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:68)
! at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:98)
! at io.dropwizard.testing.DropwizardTestSupport.startIfRequired(DropwizardTestSupport.java:325)
! at io.dropwizard.testing.DropwizardTestSupport.before(DropwizardTestSupport.java:225)
! at io.dropwizard.testing.junit5.DropwizardAppExtension.before(DropwizardAppExtension.java:195)
! at io.dropwizard.testing.junit5.DropwizardExtensionsSupport.beforeAll(DropwizardExtensionsSupport.java:97)
! at io.dropwizard.testing.junit5.DropwizardExtensionsSupport.beforeAll(DropwizardExtensionsSupport.java:83)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$10(ClassBasedTestDescriptor.java:381)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:381)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:205)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:80)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
! at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
! at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
! at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
! at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
! at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
! at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
! at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
! at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
! at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
! at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
! at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
! at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
! at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
! at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
! at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
! at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
! at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
! at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.605 s <<< FAILURE! - in io.dropwizard.client.DropwizardSSLConnectionSocketFactoryTest
[ERROR] io.dropwizard.client.DropwizardSSLConnectionSocketFactoryTest  Time elapsed: 2.605 s  <<< ERROR!
java.io.IOException: exception decrypting data - javax.crypto.BadPaddingException: pad block corrupted
```
</details>

Refs #4259
Refs #4260

--- a/dropwizard-client/src/test/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactoryTest.java
+++ b/dropwizard-client/src/test/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactoryTest.java
@@ -64,7 +64,7 @@ class DropwizardSSLConnectionSocketFactoryTest {
     }
 
     static {
-        Security.addProvider(new BouncyCastleProvider());
+        Security.insertProviderAt(new BouncyCastleProvider(), 1);
     }
 
     @AfterAll

--- a/dropwizard-client/src/test/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactoryTest.java
+++ b/dropwizard-client/src/test/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactoryTest.java
@@ -15,8 +15,6 @@ import org.glassfish.jersey.client.ClientResponse;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.condition.DisabledForJreRange;
-import org.junit.jupiter.api.condition.JRE;
 import org.junit.jupiter.api.extension.ExtendWith;
 
 import javax.net.ssl.HostnameVerifier;
@@ -42,7 +40,6 @@ import static org.assertj.core.api.Assertions.catchThrowable;
 import static org.bouncycastle.jce.provider.BouncyCastleProvider.PROVIDER_NAME;
 
 @ExtendWith(DropwizardExtensionsSupport.class)
-@DisabledForJreRange(min = JRE.JAVA_17)
 class DropwizardSSLConnectionSocketFactoryTest {
     private TlsConfiguration tlsConfiguration;
     private JerseyClientConfiguration jerseyClientConfiguration;

